scripts/update_tags_per_languages.py

```diff
@@ -0,0 +1,365 @@
+"""
+This Python code: 
+ - fetch unknown tags for categories, allergens, etc.
+ - check if each unknown tag exists in the taxonomy in different language
+ - if yes, to all products having the tag in wrong language:
+    - add existing tag in different language 
+    - remove tag in wrong language 
+
+- To use a virtual environment (depend on the OS:
+  https://python.land/virtual-environments/virtualenv)
+```python3.xx -m venv venv```
+```source venv/bin/activate```
+```pip install requests```
+
+Need to update user_id and password variables before to run the code.
+
+dev mode (dev = True) will:
+ - run in .net environment
+ - iterate over all tags until one unknown tag is found in the taxonomy (a)
+   for a different language.
+ - print possible_new_tag, already_referenced_tags, possible_wrong_language_tags
+ - iterate over a single product of all products having the tag corresponding to (a)
+ - will NOT update the products
+
+- to run tests
+```python3 -m unittest update_tags_per_languages_tests.py```
+
+
+The url used to update products should be updated to corresponding country. 
+Otherwise the language of the tags (example: categories_lc) will be updated to "en"
+Before to update all products, comment the following part
+    post_call_url_res = requests.post(
+        post_call_url.format(country=country),
+        data=data,
+        headers=headers,
+    )
+    if post_call_url_res.status_code != 200:
+        print(f"ERROR: when updating product {product['code']}. {post_call_url_res.status_code} status code")
+        sys.exit()
+and run the code, to be sure that each <tag_plural>_lc of all products 
+that should be updated are having a corresponding country in the dictionary 
+mapping_languages_countries. If not, it should print similar message:
+    "ERROR: when updating product 8850718818921. Unknown country for this language: th"
+"""
+
+import requests
+import os
+import re
+import sys
+
+# define parameters
+map_tags_field_url_parameter = {
+    # tag field in API, file name in taxonomies: url parameter
+    # "allergens": "allergen",
+    # "brands": "brand", # all unknown for now (no taxonomy for it)
+    "categories": "category",
+    # "countries": "country",
+    # "labels": "label",
+    # "origins": "origin", 
+}
+
+dev = True
+if dev:
+    env = "net"
+    user = "off:off@" 
+else:
+    env = "org"
+    user = "" 
+
+headers = {
+    'Accept': 'application/json', 
+    'User-Agent': 'UpdateTagsLanguages',
+}
+data = {
+    'user_id': '',
+    'password': '',
+}
+
+logs_file = "update_tags_per_languages_logs"
+if os.path.isfile(logs_file):
+    os.remove(logs_file)
+
+mapping_languages_countries = {
```

Comment:
thanks

left in the hashmap would be language_code in your url
right in the hashmap would be country_code_2 in your url

That would work for Denmark, ok
Not sure for Canada (language_code = "en,fr").
That would not work for "ar", "en", "xx"

I dont know if I could use your url. But it could be good if I could use it, otherwise if we find a language not listed in this hashmap while running the script my script would crash.

Reply:
It is not ideal. 
"aa", "br" (breton, french regional lang) would be missing, "xx" too -> would need to add them on top of the extraction from the url...
"ar" is special case in current dict
there are some languages (en, fr, es, etc.) that are officials in more than one country. The mapping can results in strange mapping like fr -> mg, en -> ca, es -> ni, de -> ch, etc. Not sure that it would be problematic, but would need to test it.
"en" should be replaced from whatever value to "world"


I am afraid that it would result in more lines of code. The actual code raise an error in the log. I do believe that it would be enough for a first version of the script.

Reply:
ok, that was just a suggestion ! No problem.